src/tap.h

```diff
@@ -93,8 +122,16 @@ class TestResult {
 
   std::string toString() const {
     std::stringstream ss;
-    ss << this->status << " " << this->number << " " << this->name << " "
-       << this->getComment();
+    ss << this->status << " " << this->number << " " << this->name;
+#ifdef GTEST_TAP_13_DIAGNOSTIC
+    std::string comment_text = this->getComment();
+    if (!comment_text.empty()) {
+      ss << std::endl
+       << "# Diagnostic" << std::endl
+       << "  ---" << std::endl
+       << "  " << replace_all_copy(this->getComment(), "\n", "\n  ");
+    }
+#endif
```

Comment:
I didn't test it, but isn't it still changing the default behaviour? I think it is missing a #else with a call to this->getComment()?